### ISSUE TITLE
Live Editor should accept template literals

### ISSUE THREAD

**Issue by kingcodefish** [Khan/live-editor#629]
Currently, if you use a template literal in the live editor Oh Noes will display an error like so:
**Unexpected \`.**

Here's an example: https://www.khanacademy.org/computer-programming/template-literal-test/5513940759543808

Instead, it should accept the template literal syntax without a parser error and instead spit out a warning similar to other ones displayed when ES6 features are being used. e.g.:
```
'template literal syntax (``)' is only available in ES6 which is not supported by this environment.
```

**Comment by kevinbarabash** [comment]
Currently we only support ES5.  It would be nice to support ES6 at some point.

**Comment by GoToLoop** [comment]
At least you should liberate `class` keyword.  
Much easier and less boilerplate than doing OOP w/ annexing methods by _prototype_.

**Comment by kingcodefish** [comment]
@kevinbarabash Yes, but it shouldn't produce a parser error, but rather a warning to the user of using ES6.

**Comment by SpongeJr** [comment]
Since the curriculum doesn't currently use or teach these or other ES6 concepts or design patterns, these things would benefit a very small segment of learners right now, and they are also like to be of a sort that would already be using an environment with more features, or otherwise have an alternative.

It's awesome that there are learners wanting to do more advanced things, and things outside the curriculum, but I don't think there are resources to support those changes right now.

It might be useful to file these things under a general "Support ECMAScript 6" issue tagged with "idea", though, for the distant future, or something of that sort, if @kevinbarabash sees fit.

**Comment by GoToLoop** [comment]
ES6 isn't advanced. ES6 classes are miles easier than ES5 OOP prototype pattern.
Let's hope ES6 isn't a distant future for Khan's JS.

**Comment by thenortonsetup** [comment]
Why i am not facing such an issue. I found with no issue, tech life is so boring.. No challenge at all. :(

**Comment by kevinbarabash** [comment]
The assumption is that most people using CS on khanacademy.org will be going through the tutorials which never show any ES6 syntax.  I think the current error messages are sufficient for the time being.  Also, if we were more specific in the error message it might be more confusing for beginner programmers as neither `template literal syntax` or `ES6` is mentioned in any of the tutorials.

**Comment by kingcodefish** [comment]
Then why give errors about ES6 altogether? You guys already give it errors about arrow functions, classes, function generators, etc. I've tested it. The only reason template literals are any different is because it contains a new character. My point is, make the template literal errors as clear as all the other ES6 errors.

Also, I beg to differ that beginners will never touch ES6 because it's not contained within the tutorials. A lot of beginners search the internet for code specific to whatever problem they're having. You can also see this manifest itself in certain people copying entire programs off CodePen. I know a certain number of people on there that had wanted to try out ES6 at one point and was halted by KA's errors. What if they try template literals first? All they would see is *Unexpected `.* No helpful error to them, just a syntax error. This could be very frustrating had that person not known that the environment didn't support ES6. They may try looking up what they did wrong only to never find an answer on if their template literal code is correctly written.

So, personally, I think it is logical to include the template literal error. After all, it's likely a very simple fix.

**Comment by GoToLoop** [comment]
> ... that the environment didn't support ES6.

More likely that the environment suppresses ES6 by purpose in order to allow folks w/ old abandoned browsers like IE to run all the posted codes in detriment to innovation.

**Comment by 1point7point4** [comment]
> in order to allow folks w/ old abandoned browsers like IE to still run all the posted codes in detriment to innovation

That would make sense, except...
* Polyfills, shims, and the like exist, and aren't half as complicated as some of the craziness found in the editor.
* Many browsers already don't run some code properly (#712), and nobody's trying to fix that.

I have no idea what to conclude.
